  #https://classic.gazebosim.org/tutorials?tut=ros_control

   # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50

  # Position Controllers ---------------------------------------
  shoulder_sweep_position_controller:
    type: effort_controllers/JointPositionController
    joint: shoulder_sweep_joint
    pid: {p: 1000.0, i: .001, d: 60.0, i_clamp_min: -100.0, i_clamp_max: 100.0}

  shoulder_lift_position_controller:
    type: effort_controllers/JointPositionController
    joint: shoulder_lift_joint
    pid: {p: 3000.0, i: 100.0, d: 175.0, i_clamp_min: -400.0, i_clamp_max: 400.0}

  elbow_position_controller:
    type: effort_controllers/JointPositionController
    joint: elbow_joint
    pid: {p: 2500.0, i: 100.0, d: 100.0, i_clamp_min: -100.0, i_clamp_max: 100.0}

  wrist_pitch_position_controller:
    type: effort_controllers/JointPositionController
    joint: wrist_pitch_joint
    pid: {p: 900.0, i: 10.0, d: 40.0, i_clamp_min: -400.0, i_clamp_max: 400.0}

  wrist_roll_position_controller:
    type: effort_controllers/JointPositionController
    joint: wrist_roll_joint
    pid: {p: 900.0, i: 0.1, d: 10.0, i_clamp_min: -100.0, i_clamp_max: 100.0}

  wrist_yaw_position_controller:
    type: effort_controllers/JointPositionController
    joint: wrist_yaw_joint
    pid: {p: 100.0, i: 0.1, d: 10.0, i_clamp_min: -100.0, i_clamp_max: 100.0}